src/Twig/Extension/SonataAdminExtension.php

```diff
@@ -421,6 +429,65 @@ public function getXEditableChoices(FieldDescriptionInterface $fieldDescription)
         return $xEditableChoices;
     }
 
+    /**
+     * Returns a canonicalized locale for "moment" NPM library,
+     * or `null` if the locale's language is "en", which doesn't require localization.
+     *
+     * @return null|string
+     */
+    final public function getCanonicalizedLocaleForMoment(array $context)
+    {
+        $locale = strtolower(str_replace('_', '-', $context['app']->getRequest()->getLocale()));
+
+        // "en" language doesn't require localization.
+        if (('en' === $lang = substr($locale, 0, 2)) && !in_array($locale, ['en-au', 'en-ca', 'en-gb', 'en-ie', 'en-nz'], true)) {
+            return null;
+        }
+
+        // `moment: ^2.8` only ships "es" and "es-do" locales for "es" language
+        if ('es' === $lang && !in_array($locale, ['es', 'es-do'], true)) {
+            $locale = 'es';
+        }
+
+        // @todo: there are more locales which are not supported by moment and they need to be translated/normalized/canonicalized here
```

Comment:
do we really need this comment?

Reply:
I think yes @OskarStark, since this method isn't getting rid of them (see https://github.com/sonata-project/SonataAdminBundle/pull/4878#pullrequestreview-87149336 by instance). I think these matches should be added by users which know the right locale fallback in each case.